### PR TITLE
feat: Add utilities for dealing with common inputs

### PR DESCRIPTION
**Pull request by Rhuagh**
After #757.

---
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/759)

### CODE REVIEW

File: examples/appendix_a/pong.rs [hunk 1]
```diff
@@ -27,19 +28,10 @@ impl<'a, 'b> State<GameData<'a, 'b>> for Pong {
...
-                            virtual_keycode: Some(VirtualKeyCode::Escape),
-                            ..
-                        },
-                    ..
-                } => Trans::Quit,
-                _ => Trans::None,
-            },
-            _ => Trans::None,
+        if is_close_requested(&event) || is_key(&event, VirtualKeyCode::Escape) {
+            Trans::Quit
```
Comment (Xaeroxe) [inline examples/appendix_a/pong.rs]: :heart:

**Comment by Rhuagh** [comment]
@torkleyy Fixed.

**Comment by Rhuagh** [comment]
With this kind of pattern, it will also be much easier to replace the event type that we send to the states, we just change these functions to take that event type instead and user code should still work for the most part.

**Comment by Rhuagh** [comment]
And rebased, ready to go.

**Comment by Rhuagh** [comment]
bors r=Xaeroxe,torkleyy

**Pull request merged**